UserDetails/forms.py

```diff
@@ -52,12 +57,141 @@ class Meta:
+                self.initial = self.membership.is_member()
+                if self.membership.get_verified_state() is None:
+                    self.pending = True
+
+                # Check how recently the member has been verified or not. If too recent, block change
+                if self.membership.verified_on is not None:
+                    if self.membership.is_verified:
+                        if self.membership.verified_on + \
+                                settings.DURATION_AFTER_MEMBERSHIP_CONFIRMATION > timezone.now():
+                            # The user has been verified to recently (prevent spamming)
+                            self.disabled = True
+                    else:
+                        if self.membership.verified_on + \
+                                settings.DURATION_AFTER_MEMBERSHIP_REJECTION > timezone.now():
+                            # The user has been verified not to be a member to recently (prevent spamming)
+                            self.disabled = True
+
+            except ObjectDoesNotExist:
+                pass
+        if association is None:
+            raise ValueError("Association can not be None")
+
+    def verified(self):
+        if self.membership is None:
+            return None
+        return self.membership.get_verified_state()
+
+    def get_membership_model(self, user=None, new_value=True):
+        # Check input data for correctness
+        if self.user is None and user is None:
+            raise ValueError("Field does not contain user and user was not given in method")
+        if user is not None and self.user is not None and self.user != user:
+            raise ValueError("Given user differs from field user")
+
+        if self.membership is not None:
+            return self.membership
+        if self.user is not None:
+            # If there was a user given, but the link was not found. Create a new link if allowed
+            if new_value:
+                return UserMembership(related_user=self.user, association=self.association)
+        else:
+            # user originally not given. Try to find the link
+            try:
+                return self.association.usermembership_set.get(related_user=user)
+            except ObjectDoesNotExist:
+                if new_value:
+                    return UserMembership(related_user=user, association=self.association)
+        return None
+
+
+class AssociationLinkForm(forms.Form):
+
+    def __init__(self, user, *args, **kwargs):
+        super(AssociationLinkForm, self).__init__(*args, **kwargs)
+
+        self.user = user
+
+        if user is None:
+            associations = Association.objects.filter(is_choosable=True)
+        else:
+            associations = Association.objects.filter(
+                Q(is_choosable=True) |
+                (Q(is_choosable=False) & Q(usermembership__related_user=user))) \
+                .distinct().order_by('slug')
```

Comment:
I don't think so either, but I do know that it originally was added because it was needed, though I may have altered the exact query contents since then. I'll have to double check this.